### PR TITLE
[5.1]docs(contextmenu): item render event documentation

### PR DESCRIPTION
**Pull request by YanushevGeorgi**
Related to: [#8166](https://github.com/telerik/blazor/issues/8166)

### CODE REVIEW

File: components/contextmenu/events.md [hunk 1]
```diff
@@ -10,9 +10,93 @@ position: 30
 
 # Events
 
-This article explains the events available in the Telerik Context Menu for Blazor:
+This article describes the events available in the Telerik Context Menu for Blazor:
```
Comment (yordan-mitev) [inline components/contextmenu/events.md:13]:
```suggestion
This article describes the events generated by the Telerik Context Menu for Blazor:
```

**Review by yordan-mitev — APPROVED** [review]

File: components/contextmenu/events.md [hunk 2]
```diff
@@ -10,9 +10,93 @@ position: 30
 
 # Events
 
-This article explains the events available in the Telerik Context Menu for Blazor:
+This article describes the events generated by the Telerik Context Menu for Blazor:
```
Comment (dimodi) [inline components/contextmenu/events.md:13]:
```suggestion
This article describes the events that are fired by the Telerik Context Menu for Blazor:
```

File: components/contextmenu/events.md [hunk 3]
```diff
@@ -10,9 +10,93 @@ position: 30
...
 
-* [OnClick](#onclick)
+* [`OnClick`](#onclick)
+* [`OnItemRender`](#onitemrender)
```
Comment (dimodi) [inline components/contextmenu/events.md:16]: The events are sorted alphabetically, which is OK, but then the sections are not.

File: components/contextmenu/events.md [hunk 4]
```diff
@@ -10,9 +10,93 @@ position: 30
...
+The `OnItemRender` event fires when each Context Menu item renders. It allows you to customize the appearance of an item.
+
+The event handler receives an argument object of type `MenuItemRenderEventArgs` that contains the following properties: 
+
+| Property | Type | Description |
+| --- | --- | --- |
+| `Item` | `object` | The current item that renders in the Context Menu. |
+| `Class` | `string` | The custom CSS class that will be added to the item. |
+
+>caption Customizing the appearance of the Context Menu items.
```
Comment (dimodi) [inline components/contextmenu/events.md:29]:
```suggestion
>caption Using OnItemRender to customize the appearance of the Context Menu items.
```

File: components/contextmenu/events.md [hunk 5]
```diff
@@ -10,9 +10,93 @@ position: 30
...
+
+````CSHTML
+<div class="context-menu-target" style="width:200px; height: 100px; background: yellow; margin-bottom: 50px;">
+    Right-click (or tap and hold on a touch device) for a Context Menu.
+</div>
+
+<TelerikContextMenu Data="@MenuItems" Selector=".context-menu-target" OnItemRender="@RenderItemHandler">
```
Comment (dimodi) [inline components/contextmenu/events.md:36]: Consider multi-line declaration

File: components/contextmenu/events.md [hunk 6]
```diff
@@ -10,9 +10,93 @@ position: 30
...
+</style>
+
+@code {
+    public List<ContextMenuItem> MenuItems { get; set; }
+
+    private void RenderItemHandler(MenuItemRenderEventArgs args)
+    {
+        var item = args.Item as ContextMenuItem;
+
+        args.Class = "orange";
```
Comment (dimodi) [inline components/contextmenu/events.md:52]: Use a conditional statement to style only 1-2 items, or use several classes and colors.

**Review by dimodi — CHANGES_REQUESTED** [review]
* Sort event sections alphabetically
* Style one item only, or use different colors

File: components/contextmenu/events.md [hunk 7]
```diff
@@ -87,6 +88,93 @@ You can use the `OnClick` event to react to user choices, for example load new c
...
+    .orange {
+        background: orange;
+    }
+    .lime {
+        background: lime;
+    }
+</style>
+
+@code {
+    public List<ContextMenuItem> MenuItems { get; set; }
```
Comment (ntacheva) [inline components/contextmenu/events.md:124]:
```suggestion
    private List<ContextMenuItem> MenuItems { get; set; }
```
According to code style

File: components/contextmenu/events.md [hunk 8]
```diff
@@ -87,6 +88,93 @@ You can use the `OnClick` event to react to user choices, for example load new c
...
+</style>
+
+@code {
+    private List<ContextMenuItem> MenuItems { get; set; }
+
+    private void RenderItemHandler(MenuItemRenderEventArgs args)
+    {
+        var item = args.Item as ContextMenuItem;
+
+        args.Class = item.Items==null ? "orange" : "lime";
```
Comment (dimodi) [inline components/contextmenu/events.md:130]:
```suggestion
        args.Class = item.Items == null ? "orange" : "lime";
```

**Review by dimodi — APPROVED** [review]

**Pull request merged**